docs/docs/integrations/promotions/voucherify.md

```diff
@@ -0,0 +1,119 @@
+
+As we see above, the cart already has basic services responsible for handling discounts.
+However, in order to use Voucherify as a tool for managing vouchers and promotions, it was necessary to create functions that would use Voucherify REST API.
+
+To extend the storefront by capabilities provided by Voucherify integration a separate package called `voucherify` has been created, which contains the necessary functions needed to manage discounts.
+
+`@composable/voucherify` dependency implements a standard checkout integration pattern, where for each cart update, the Voucherify functions make Validation (eligibility check, discount calculations) and Qualification (list of applicable promotions) requests to Voucherify REST API. Collected pieces of information extend cart data.
+
+```mermaid
+sequenceDiagram
+  participant B as Browser
+  participant CV as @composable/voucherify
+  participant CGS as Commerce Generic
+  participant V as Voucherify API
+  B->>CGS: Discount related methods (addVoucherToCart, updateCartDiscount etc.)
+    activate CV
+  CGS->>CV: Use Voucherify specific methods from @composable/voucherify
+  activate CGS
+  CGS-->>CV: Cart object
+  deactivate CGS
+  Note over CV: Get cart vouchers codes from storage
+  CV->>V: Get available cart promotions
+  activate V
+  V-->>CV: promotions
+  deactivate V
+  CV->>V: Validate promotions and voucher codes
+  activate V
+  V-->>CV: Eligibility check and discount calculations results
+  deactivate V
+  Note over CV: Extend cart by discounts details
+  CV-->>B: Cart object containing discounts from Voucherify
+  deactivate CV
+```
+
+## Reference Files
+
+### Backend Files
+
+- `packages/voucherify`: This package contains functions that enhance the default behavior of functions related to discounts.
+
+### React Components
+
+- `composable-ui/src/components/cart/cart-vouchers.tsx`: A component that allows you to add, display, validate and delete vouchers.
+- `composable-ui/src/components/forms/voucher-form.tsx`: Form used in `cart-vouchers.tsx`.
+- `composable-ui/src/components/cart/cart-promotions.tsx`: Displays auto-applied cart promotions.
+
+## Integrating Voucherify with Composable UI
+
+### First steps
+
+1. [Create a Voucherify account](https://app.voucherify.io/#/signup).
+2. In Voucherify Dashboard, [set Discount Application Rule to "Partial"](https://support.voucherify.io/article/604-stacking-rules#application-rules)
+   Please go to:
+- `Redemptions` tab on the left side of the Voucherify dashboard
+- Select `Stacking rules`
+- In `Discounts' application rules` select `Edit settings`
+- Change the `Application rule` to `Partial - inapplicable (not met criteria) discounts are omitted`
+3. Retrieve your API keys from your Voucherify dashboard and set the following environment variables:
+
+```code
```

Comment:
Please add a note about adding these env vars to both the `composable-ui` app as well as in `scripts/.env`